oozie-to-airflow/mappers/shell_mapper.py

```diff
@@ -0,0 +1,94 @@
+# Copyright 2018 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import os
+
+import jinja2
+from airflow.utils.trigger_rule import TriggerRule
+
+from definitions import ROOT_DIR
+from mappers.action_mapper import ActionMapper
+from mappers.prepare_mixin import PrepareMixin
+from utils import el_utils, xml_utils
+
+
+class ShellMapper(ActionMapper, PrepareMixin):
+    """
+    Converts a Shell Oozie action to an Airflow task.
+    """
+
+    def __init__(
+        self,
+        oozie_node,
+        task_id,
+        trigger_rule=TriggerRule.ALL_SUCCESS,
+        params=None,
+        template="shell.tpl",
+        **kwargs,
+    ):
+        ActionMapper.__init__(self, oozie_node, task_id, trigger_rule, **kwargs)
+        if params is None:
+            params = {}
+        self.template = template
+        self.params = params
+        self.task_id = task_id
+        self.trigger_rule = trigger_rule
+        self.properties = {}
+        self._parse_oozie_node()
+
+    def _parse_oozie_node(self):
+        res_man_text = self.oozie_node.find("resource-manager").text
+        name_node_text = self.oozie_node.find("name-node").text
+        self.resource_manager = el_utils.replace_el_with_var(res_man_text, params=self.params, quote=False)
+        self.name_node = el_utils.replace_el_with_var(name_node_text, params=self.params, quote=False)
+        self._parse_config()
+        self._parse_params()
+        cmd_node = self.oozie_node.find("exec")
+        arg_nodes = self.oozie_node.findall("argument")
+        cmd = " ".join([cmd_node.text] + [x.text for x in arg_nodes])
+        self.bash_command = el_utils.convert_el_to_jinja(cmd, quote=False)
+
+    def _parse_params(self):
```

Comment:
What are those params?. I could not find it in https://oozie.apache.org/docs/5.0.0/DG_ShellActionExtension.html

Reply:
You're right, it shouldn't be here. Copy-pasting gone wrong ;) I'll remove it.

Reply:
Done